proposals/0000-merge-datepickerios-and-datepickerandroid.md

```diff
@@ -0,0 +1,229 @@
+  const Selector = mode === 'time' ? TimePickerAndroid : DatePickerAndroid;
+
+  Selector.open({ ...value, ...props }) // act on thenable with `onDateChange`
+}
+```
+
+### Naming
+
+The proposed name is `react-native-datetimepicker`, with a Github repo in the react native community 
+`git@github.com:react-native-community/react-native-datetimepicker.git`. However, suggestions are welcome!
+
+### Options 
+
+Property names should remain similar where possible. However there are some difference between android and iOS
+that need to be resolved. This section will talk about required changes.
+
+#### value
+
+Both `DatePickerIOS` and `DatePickerAndroid` use `date` whereas `TimePickerAndroid` requires an `hour` and 
+`minute` property. The proposal is to change this property to `value {Date, Number}`. Where `{Number}` is
+a unix epoch timestamp. Both [Android][month-android] and [JS][month] use a month integer from `[n](0,11)`, 
+making conversions from and to `Date` relatively easy. Also converting a `Date` to `timestamp` is simple (iOS
+requires timestamps). `Date` also supports setting the `hour` and `minute` making it an ideal candidate to
+support all native implementations.
+
+Finally, using a more generic description for `date` and `time` allows it to be [used for setting
+`countDownDuration`][countdown]. Enabling the [feature implementation of `mode: countdown`](#mode) or
+doing a [calculation of `timeZoneOffsetInMinutes` in JS](#timezoneoffsetinminutes-ios).
+
+#### onDateChange
+
+The `eventHandler` will function on iOS as before and ensure the controlled component works as expected. 
+Android will act on the returned `Promise`.
+
+```js
+<DatePickerAndroid, TimePickerAndroid>.then(function resolve({ action, day, month, year }) {
+  if (action === 'dateSetAction') onDateChange(new Date(year, month, day));
+  if (action === 'dismissedAction') onDateChange();
+}, function reject(error) {
+  // ignore or throw `activity == null` error
+});
+```
+
+Note: the [`activity == null` error rejection][reject] requires a solution. This `Error` could either be thrown to
+let it surface as red screen or ignored entirely. 
+
+#### maximumDate & minimumDate
+
+Use the `iOS` property called `maximumDate` to constraint date ranges. This implies `maxDate` from
+Android is renamed to `maximumDate`. The same approach will be used to rename the `minDate` to `minimumDate`.
+These options should be ignored for `TimePickerAndroid`.
+
+#### mode 
+
+The current `mode` should be split up in two different properties, `mode` and `display`. The default for
+`mode` should be changed to `date`. Both iOS specific options would render the default on Android. 
+Support for platform specific implementations can stay the same. For example, `mode: datetime` would be 
+ignored for Android. Available values for `mode`:
+
+- `date` (new default for both android and iOS)
+- `time`
+- `datetime (iOS)` 
+- `countdown (iOS)`
```

Comment:
Although "countdown" is technically one of the available modes of the `UIDatePicker`, I'm not sure it belongs here. Two things combine to make it weird in this context: 1) it's always iOS-only; 2) it represents a time *span*, not a point in time. How crazy would it be to create a separate `CountdownPickerIOS` RN component to encapsulate just a `UIDatePicker` in `countDownTimer` mode, with an interface to set the minutes on the countdown timer?  That would free `DateTimePicker` from having the responsibility of representing a concept that is distinct from a date or time.

Reply:
That could be a path forward, I consider this outside the scope of this proposal though. The option currently exists for native iOS but it is not fully implemented (wrt configurability). It is sort of usable right now, without it being documented completely. https://github.com/react-native-community/discussions-and-proposals/pull/85/files/3255b2c845018924c2d0a6978bcf91f92893f736#diff-cfdf3e4461139c8caebcec975f57c066R163 It is here for completeness. As it stands the new implementation should not implement specific logic around this.